substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVRootPackages.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2020, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.pecoff.cv;
+
+import java.util.Collections;
+import java.util.HashSet;
+
+abstract class CVRootPackages {
```

Comment:
This is very fragile in the face of future GraalVM development. You don't actually need to list every single Graal package just the package roots that can be guaranteed to cover all current packages (and are pretty sure to capture any future additions). If you look at GraalSourceCache it employs a much smaller set of package prefixes to determine that a class is GraalVM code (and hence needs its source located in a src.zip file rather than a jar). It's still fragile in but much less so. If all of this is really needed to allow the skipGraalInternals mode to work then I think maybe that's a good reason not to retain that mode.
The same issue arises with the list of intrinsic classes that follows.

Reply:
I know.  It hurt to write.  It's gone.